File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/javadoc/comment/direct/Java/C/s527301188_commented.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    char input[100];
    fgets(input, sizeof(input), stdin);

    char *token = strtok(input, " ");
    int nums[100];
    int top = 0;

    while (token != NULL) {
        if (token[0] == '+') {
            top--;
            nums[top-1] = nums[top - 1] + nums[top];
        } else if (token[0] == '-') {
            top--;
            nums[top-1] = nums[top - 1] - nums[top];
        } else if (token[0] == '*') {
            top--;
            nums[top-1] = nums[top - 1] * nums[top];
        } else {
            nums[top] = atoi(token);
            top++;
        }
        token = strtok(NULL, " ");
    }

    printf("%d\n", nums[0]);
    // End of Code
}

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s527301188_commented.go
package main

import (
	"bufio"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	line, _ := reader.ReadString('\n')
	line = strings.TrimSpace(line)
	tokens := strings.Split(line, " ")

	nums := make([]int, len(tokens))
	top := 0

	for _, token := range tokens {
		if len(token) == 1 && (token[0] == '+' || token[0] == '-' || token[0] == '*') {
			top--
			switch token[0] {
			case '+':
				nums[top-1] += nums[top]
			case '-':
				nums[top-1] -= nums[top]
			case '*':
				nums[top-1] *= nums[top]
			}
		} else {
			num, _ := strconv.Atoi(token)
			nums[top] = num
			top++
		}
	}
	println(nums[0])
}

// End of Code
